env/mock_env.cc

```diff
@@ -394,7 +394,8 @@ class TestMemLogger : public Logger {
       assert(p <= limit);
       const size_t write_size = p - base;
 
-      file_->Append(Slice(base, write_size));
+      Status s = file_->Append(Slice(base, write_size));
+      assert(s.ok());
```

Comment:
How about checking `s.ok()`. Only when ok do we set `flush_pending_` and increment `log_size_`?